bluepyopt/ephys/create_hoc.py

```diff
@@ -60,26 +75,60 @@ def _generate_reinitrng(mechs):
     return reinitrng_content
 
 
-def _generate_parameters(parameters):
+def _range_exprs_to_hoc(range_params):
+    """Process raw range parameters to hoc strings"""
+
+    ret = []
+    for param in range_params:
+        value = param.value_scaler.inst_distribution
+        value = re.sub(r'math\.', '', value)
+        value = re.sub('{distance}', FLOAT_FORMAT, value)
+        value = re.sub('{value}', format_float(param.value), value)
+        ret.append(Range(param.location, param.name, value))
+    return ret
+
+
+def _loc_desc(location, param_or_mech):
+    """Generate Neuron location description for HOC template"""
+
+    if isinstance(param_or_mech, mechanisms.NrnMODMechanism):
+        # TODO this is dangerous, implicitly assumes type of location
+        return getattr(location, 'seclist_name', 'all')
+    elif isinstance(param_or_mech, mechanisms.NrnMODPointProcessMechanism):
+        raise ValueError("%s is currently not supported by create_hoc." %
+                         type(param_or_mech).__name__)
+    # FIXME: NrnSectionCompLocation
```

Comment:
I don't think the FIXME annotation is problematic - this hints to a problem with the implementation that was already present in the master branch.

Reply:
Ok, I thought you wanted to fix this before merging this PR and forgot to remove the comment.
Is this problem already mentioned in the issues? If not, could you create an issue about it so that we can fix it?
Thanks

Reply:
This is the reproducing example:

```python
BluePyOpt$ ipython
Python 3.8.10 (default, Jun 22 2022, 20:18:18) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from bluepyopt import ephys

In [2]: ephys.create_hoc.create_hoc(mechs=[], parameters=[ephys.parameters.NrnSectionParameter(name='param', value=1., locations=[ephys.locations.NrnSectionCompLocation(name='loc', sec_name='soma[0]', comp_x=0.5)])])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [2], line 1
----> 1 ephys.create_hoc.create_hoc(mechs=[], parameters=[ephys.parameters.NrnSectionParameter(name='param', value=1., locations=[ephys.locations.NrnSectionCompLocation(name='loc', sec_name='soma[0]', comp_x=0.5)])])

File ~/src/arbor/bluepyopt_test/BluePyOpt/bluepyopt/ephys/create_hoc.py:156, in create_hoc(mechs, parameters, morphology, ignored_globals, replace_axon, template_name, template_filename, disable_banner, template_dir, custom_jinja_params)
    152     template = template_file.read()
    153     template = jinja2.Template(template)
    155 global_params, section_params, range_params, location_order = \
--> 156     _generate_parameters(parameters)
    157 channels = _generate_channels_by_location(mechs, location_order)
    159 ignored_global_params = {}

File ~/src/arbor/bluepyopt_test/BluePyOpt/bluepyopt/ephys/create_hoc.py:76, in _generate_parameters(parameters)
     73         assert isinstance(
     74             param.locations, (tuple, list)), 'Must have locations list'
     75         for location in param.locations:
---> 76             param_locations[location.seclist_name].append(param)
     78 section_params = defaultdict(list)
     79 range_params = []

AttributeError: 'NrnSectionCompLocation' object has no attribute 'seclist_name'
```

If `create_hoc` only supports areal locations, I think neither of the other `CompLocation`s should be accepted by `create_hoc` (`NrnSeclistCompLocation`, `NrnSomaDistanceCompLocation`, `NrnSecSomaDistanceCompLocation`, `NrnTrunkSomaDistanceCompLocation`). Effectively, it seems that only `NrnSeclistLocation` is supported for the mechanism location, hence, I've added a check. It might be worth distinguishing areal from point locations in the class hierarchy (below `locations.Location`) in the future. In Arbor this is the difference between regions and locsets.

Reply:
I just created an issue (#421) from this thread so that it's not forgotten after this PR.

Reply:
I've already addressed this (#421) as hinted above - the commit is 9bbc68ed8396a8985114d9b7b568990658cd9369. I think follow-up issues should focus on extending hoc-support or, as suggested above, introducing another level in the `locations` class hierarchy that distinguishes areal from point locations.